diff -Naur glest-source-3.2.2/shared_lib/sources/platform/posix/socket.cpp glest-source-3.2.2.tpg/shared_lib/sources/platform/posix/socket.cpp
--- glest-source-3.2.2/shared_lib/sources/platform/posix/socket.cpp	2009-04-03 17:40:41.000000000 +0000
+++ glest-source-3.2.2.tpg/shared_lib/sources/platform/posix/socket.cpp	2009-09-20 16:51:37.000000000 +0000
@@ -226,7 +226,7 @@
 	if(err < 0) {
 	    char szBuf[1024]="";
 	    sprintf(szBuf,"#2 Error connecting socket for IP: %s for Port: %d err = %d errno = %d [%s]",ip.getString().c_str(),port,err,errno,strerror(errno));
-	    fprintf(stderr,szBuf);
+	    fprintf(stderr,szBuf, "%s");
 
         if (errno == EINPROGRESS) {
 
